src/actions/Scroll.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+  SelectionWithEditor,
+} from "../Types";
+import { displaySelectionDecorations } from "../editDisplayUtils";
+import { groupBy } from "../itertools";
+import { commands, TextEditor, Selection, window } from "vscode";
+import { focusEditor } from "./setSelectionsAndFocusEditor";
+
+class Scroll implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph, private at: string) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const selectionGroups = groupBy(
+      targets,
+      (t: TypedSelection) => t.selection.editor
+    );
+
+    const selections: SelectionWithEditor[] = [];
+    const lines: LineWithEditor[] = [];
+
+    selectionGroups.forEach((targets, editor) => {
+      const { lineNumber, lineNumberToDecorate } = getLineNumber(
+        targets,
+        this.at
+      );
+      lines.push({ lineNumber, editor });
+      const s = new Selection(lineNumberToDecorate, 0, lineNumberToDecorate, 0);
+      selections.push({ selection: s, editor });
+    });
+
+    const scrollCallback = async () => {
+      const originalEditor = window.activeTextEditor;
+
+      await Promise.all(
+        lines.map(async (lineWithEditor) => {
+          // For reveal line to the work we have to have the correct editor focused
+          if (lineWithEditor.editor !== window.activeTextEditor) {
+            await focusEditor(lineWithEditor.editor);
+          }
+          await commands.executeCommand("revealLine", {
+            lineNumber: lineWithEditor.lineNumber,
+            at: this.at,
+          });
+        })
+      );
+
+      // If necessary focus back original editor
+      if (originalEditor && originalEditor !== window.activeTextEditor) {
```

Comment:
Idk you guys both seem to prefer the more compact for this one but I do tend to prefer

```suggestion
      if (originalEditor != null && originalEditor !== window.activeTextEditor) {
```